### ISSUE TITLE
Unreachable code detection does not respect local scopes and throw statements

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#75605]
## Version Used
VS 2022 Version 17.11.5
Roslyn 4.11.0 (5649376e0e5f)

Project info
C# 12.0
.NET 8.0

## Steps to Reproduce

Use the following code:

```csharp
public static int Add(int a, int b)
{
    {
        return a + b;
    }

    throw new InvalidOperationException();

    {
        throw new InvalidOperationException();
    }

    {
        const int x = 523;
        throw new InvalidOperationException();
    }

    {
        throw new InvalidOperationException();
    }
}
```

## Expected Behavior

In VS, the unreachable code will dim out showing that the dimmed out region is unreachable. In the above example, we expect all the code from the throw statement onwards to be dimmed out.

## Actual Behavior

The resulting dimming is very different and limited to the scope that starts with non-throw statements.
![Image](https://github.com/user-attachments/assets/87e0accf-733f-4dd5-ae1a-f2784308e3ac)

**Comment by CyrusNajmabadi** [comment]
Very minor issue.  Would take a community pr if someone wanted to fix it.

**Comment by CyrusNajmabadi** [comment]
i need someone from compiler to look at this.  Compiler is not reporting that the 2nd/3rd throw-statements are unreachable.  We drive classification here from the compiler diagnostics, so unless this is fixed on the compiler side, we can't do anything on the IDE side.
